### PR TITLE
Dispatch events in AccountManagement::checkPasswordStrength()

### PR DESCRIPTION
**Pull request by tdgroot**
Dispatch events in AccountManagement::checkPasswordStrength() to improve extensibility.

### Manual testing scenarios

1. Create an observer that hooks in on `customer_check_password_strength_before` or `customer_check_password_strength_after`.
2. Create a customer, the observer should be called and the password provided as well.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by orlangur** [comment]
Hi @tdgroot, it is not recommended to add any new events in general, plugins-interceptors should be implemented whenever possible instead of event observers.

If current code does not allow to write plugin-interceptor in some case, it should be refactored so that intercepting is possible.

Please provide real-world customization scenario you trying to cover with these new events.

**Comment by tdgroot** [comment]
@orlangur I considered making the method public, but that would mean it actually has to be added to the interface. 

The way I see it, there are 3 options:
- Make the method `checkPasswordStrength()` public.
- Create and additionally call a new (public) method to allow for extension possibilities.
- Create a password validator class to keep the password validation logic in that single class. This also allows for creation of new public methods and new extension possibilities.  

What's your suggestion in this case? 

p.s. as for the real-world customization scenario, I have written a pwned password validator for Magento 2. Unfortunately I had to rewrite the AccountManagement class to create an extension point to add my validation. The module: [magento2-module-pwned-validator](https://github.com/tdgroot/magento2-module-pwned-validator)

**Comment by orlangur** [comment]
@tdgroot oh, nice, I didn't expect to see an existing implementation 👍 

To me it seems the best to move `checkPasswordStrength` method logic into some new `PasswordStrengthChecker` class.

**Comment by tdgroot** [comment]
@orlangur I created a little implementation of the logic inside classes. Please let me know if this is going to work for Magento. If it is, I can start writing tests for the logic.

File: app/code/Magento/Customer/Model/PasswordStrengthChecker.php
Comment (Jeroenll) [inline app/code/Magento/Customer/Model/PasswordStrengthChecker.php]: Can we mark this method private or remove entirely since it's only used in `check()`?

Comment (tdgroot) [reply app/code/Magento/Customer/Model/PasswordStrengthChecker.php]: Spot on! I'll make it private.

**Comment by tdgroot** [comment]
@orlangur what do you think?

**Comment by orlangur** [comment]
@tdgroot will review shortly, no need to fix tests yet - let me check implementation first.

**Review by orlangur — CHANGES_REQUESTED** [review]
Nowadays all changes must be applied to `2.3-develop` first (see https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules). Please prepare a new pull request, this one should be `on hold` until the latter is merged.

**Comment by orlangur** [comment]
Closing due to inactivity. Feel free to reach me out anytime later if you wish to continue work on this pull request and it will be reopened.
Thanks for collaboration!

**Pull request closed without merging**
